TypeInjections/Tool/Tool.fs

```diff
@@ -57,41 +64,52 @@ module Tool =
             checkoutCommit commit1 location pathOld
             checkoutCommit commit2 location pathNew
 
+            pathOld, pathNew, $"{path}/Output"
 
-        // -r [github repo url] [github commit hash] [github commit hash] [OutputPath]
-        elif flag.Contains("-r") then
-
-            Console.WriteLine("***** remote github")
-
-            if argvList.Length <> 5 then
-                failwith "usage: program -r GITHUB_URL OLD_COMMIT NEW_COMMIT OUTPUTFOLDER"
-
-            let url = argvList.Item(1)
+        else if not remote.IsEmpty then
+            Utils.log "***** remote github"
 
-            let commit1 = argvList.Item(2)
-            let commit2 = argvList.Item(3)
-
-            let path = argvList.Item(4)
+            let path = remote.Item(3)
             let pathOld = $"{path}/Old"
             let pathNew = $"{path}/New"
+            let url = remote.Item(0)
+            let commit1 = remote.Item(1)
+            let commit2 = remote.Item(2)
 
-            // lib2git requires an empty folder to place repos
             for a in [ pathOld; pathNew ] do
                 if Directory.Exists(a) then
                     Directory.Delete(a, true)
 
-            // grab repos of new and old projects
             checkoutCommit commit1 url pathOld
             checkoutCommit commit2 url pathNew
 
-        else
-            failwith "flag not found please try again"
 
+            pathOld, pathNew, $"{path}/Output"
+
+        else
+            failwith "unable to read flags"
+            
+    // organize arguments with the flag they associated with 
+    let run opt =
+        let local = opt.local |> Seq.toList
+        let git = opt.git |> Seq.toList
+        let remote = opt.remote |> Seq.toList
+        // given the list of arguments associated with each flag, run typeCart
+        let oldFolder, newFolder, outputFolder = interpretFlags local git remote
+
+        Utils.log $"old folder is {oldFolder}, new folder is {newFolder}, output folder is {outputFolder}"
 
     [<EntryPoint>]
     let main (argv: string array) =
 
         Utils.log "***** Entering Tool.fs"
-        let argvList = argv |> Array.toList
-        interpretFlag argvList
+
+        let result =
+            Parser.Default.ParseArguments<options>(argv)
+
+        match result with
+        | :? (Parsed<options>) as parsed -> run parsed.Value
+        | :? (NotParsed<'a>) as notParsed -> ()
+
```

Comment:
These warnings are worthy of investigation. What's the type of `result`? You might want a `| _ -> ... ` catch all case, but not sure what's the right thing to do here.

Reply:
I agree, for now I will add `| _ -> ...` and next PR I will handle them with more logic